extensions/wikia/WikiaMobile/maintenance/ConfigureSmartBanner.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * @author Mix (Michał Roszka) <mix@wikia-inc.com>
+ * @ingroup Maintenance
+ */
+require_once __DIR__ . '/../../../../maintenance/Maintenance.php';
+
+error_reporting(E_ALL);
+ini_set('display_errors', 1);
+
+class ConfigureSmartBanner extends Maintenance {
+
+	const REASON_ENABLED = 'Enabled Wikia Mobile Apps smart banner.';
+	const REASON_DISABLED = 'Disabled Wikia Mobile Apps smart banner.';
+
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription = 'Configure Wikia Mobile Apps smart banner.';
+
+		// Disable the smart banner completely.
+		$this->addOption( 'disable', 'Disable the banner', false, false);
+		// Enable the banner (all the options  must have arguments)
+		$this->addOption( 'icon', 'Icon URL', false, true );
+		$this->addOption( 'title', 'Title for the banner', false, true );
+		$this->addOption( 'ios-app-id', 'iOS App ID', false, true );
+		$this->addOption( 'android-app-id', 'Android Application Package ID', false, true);
+	}
+
+	public function execute() {
+		$wikiId = F::app()->wg->cityId;
+
+		if ( $this->getOption('disable') ) {
+			// Try to disable the banner.
+			WikiFactory::setVarByName( 'wgEnableWikiaMobileSmartBanner', $wikiId, false, self::REASON_DISABLED )
+				or $this->error( 'Failed to set wgEnableWikiaMobileSmartBanner.', 1 );
+			$this->output( "Disabled smart banner for $wikiId.\n" );
+			exit(0);
+		}
+
+		if ( ! $this->getOption('title') || ! $this->getOption('icon') || ! $this->getOption('ios-app-id') || ! $this->getOption('android-app-id') ) {
+			$this->error( 'Required parameters: title, icon, ios-app-id, android-app-id', 1 );
+		}
+		
+		// Banner configuration.
+		$config = [
+			'disabled'	=> 0,
+			'name'		=> "Wikia: {$this->getOption( 'title' )} Fan App",
+			'icon'		=> $this->getOption( 'icon' ),
+			'meta'		=> [
+						'apple-itunes-app'	=> "app-id={$this->getOption( 'ios-app-id' )}",
+						'google-play-app'	=> "app-id={$this->getOption( 'android-app-id' )}",
+			],
+		];
+
+		// Try to set the banner configuration first.
+		WikiFactory::setVarByName( 'wgWikiaMobileSmartBannerConfig', $wikiId, $config, self::REASON_ENABLED )
+			or $this->error( 'Failed to set wgWikiaMobileSmartBannerConfig.', 1 );
```

Comment:
`$wgUser` can be set to `WikiaBot` at the top of the script (or any other existing user account that makes sense here). WikiFactory log entries then be marker as made by this user. Can be handy for folks reading these logs.